kfp_notebook/pipeline/_notebook_op.py

```diff
@@ -228,6 +228,18 @@ def __init__(self,
             gpu_vendor = self.pipeline_envs.get('GPU_VENDOR', 'nvidia')
             self.container.set_gpu_limit(gpu=str(gpu_limit), vendor=gpu_vendor)
 
+        # Attach metadata to the pod
+        # Node type (a static type for this op)
+        self.add_pod_label('elyra-node-type', 'notebook-script')
+        # Pipeline name
+        # Should we use kwargs.get('pipeline_name')?
+        self.add_pod_label('elyra-pipeline-name',
+                           self.cos_directory)
+        # Pipeline node name
+        self.add_pod_label('elyra-node-name', kwargs.get('name'))
```

Comment:
I forgot that we need to sanitize user-provided input to make sure it adheres to the _value_ constraints Kubernetes imposes. Do we already have a general purpose utility "class" that I could augment with two methods that convert a string to a valid kubernetes label/annotation value?
The keys I'm using are static and therefore don't require any processing.

Reply:
Good point.  I doubt anything like that exists.  However, [annotation values](https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/#syntax-and-character-set) do not adhere to the same constraints as [label values](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set).

> If the prefix is omitted, the annotation Key is presumed to be private to the user. **Automated system components** (e.g. kube-scheduler, kube-controller-manager, kube-apiserver, kubectl, or **other third-party automation) which add annotations to end-user objects must specify a prefix**.

Seems like we should treat `elyra` as the prefix portion for the label and annotation keys: e.g., `elyra/node-node`, `elyra/pipeline-name`

Reply:
Done
 - Added a static class method that validates the label value. No validation performed for annotation values because I couldn't find a formal definition of what constraints are enforced.
 - Used `elyra` prefix as suggested, although it'd be better if there was an `elyra.io` (or similar) DNS entry that the elyra project owns, that ties the prefix to something people can look up